### PR TITLE
Fix constraints tests

### PR DESCRIPTION
**Pull request by alneberg**
Added tests to verify the fix_constraints PR. Everything worked as it should except project deletion which is here fixed by adding `nullable=False` to versions and files.

NOTE, this PR also changes the github actions tests to run on any PR, not just against dev.

### CODE REVIEW

File: tests/test_db_constraints.py [hunk 1]
```diff
@@ -0,0 +1,569 @@
...
+
+    for file in project.files:
+        db.session.delete(file)
+
+    db.session.delete(project)
+
+    db.session.commit()
+
+
+def test_delete_unit_row__with_project_and_users(client):
```
Comment (inaod568) [inline tests/test_db_constraints.py:46]: Already mentioned but adding here too: 
Make sure to test with empty projects, currently the projects contain files.

File: tests/test_db_constraints.py [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+    Error
+        Need to delete UnitUsers (due to inheritance issues) rows first
+    """
+    unit, projects, _ = __setup_unit()
+
+    for project in projects:
+        __delete_project(project)
+
+    # Removing this print statement causes the test to fail on Johannes computer. I don't know why.
+    print(unit.users)
```
Comment (inaod568) [inline tests/test_db_constraints.py:73]: Does it still do that?

File: tests/test_db_constraints.py [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+    Error
+        Need to delete File rows and Version rows first
+    """
+    project = __setup_project()
+
+    for file in project.files:
+        db.session.delete(file)
+    db.session.commit()
+
+    version_ids = [version.id for version in project.file_versions]
```
Comment (inaod568) [inline tests/test_db_constraints.py:179]: What's this used for?

File: tests/test_db_constraints.py [hunk 4]
```diff
@@ -0,0 +1,569 @@
...
+
+    db.session.delete(file)
+    db.session.commit()
+
+    exists = models.File.query.filter_by(name=filename).one_or_none()
+    assert exists is None
+
+    exists = models.Project.query.filter_by(id=project_id).one_or_none()
+    assert exists is not None
+    print(file.id)
```
Comment (inaod568) [inline tests/test_db_constraints.py:525]: Do we want this here?

Comment (inaod568) [reply tests/test_db_constraints.py:525]: Just wondering if it's just a debugging thing for now, if it's the error on your laptop again, or if we can remove it

Comment (alneberg) [reply tests/test_db_constraints.py:527]: Just debugging left by mistake, will fix!

File: tests/test_db_constraints.py (hunk 3 above)
Comment (alneberg) [reply tests/test_db_constraints.py:181]: Nothing! Will delete.

File: tests/test_db_constraints.py (hunk 2 above)
Comment (inaod568) [reply tests/test_db_constraints.py:75]: Great! 👏🏻

**Review by inaod568 — APPROVED** [review]

**Pull request merged**